### PR TITLE
Add Shutdown All handling

### LINKED ISSUE #3467: Add "Shutdown all Kernels" command and UI
A common use case for the Running panel is to shutdown all kernels. We should provide a command to shutdown all kernels and:

* A "Shutdown all Kernel" menu item in the "Kernel Menu"
* A command palette entry
* A button to shutdown all kernels in the Running tab

### PR DESCRIPTION
**Pull request by blink1073**
Fixes #3467.

image

image

image

image

image

### CODE REVIEW

**Comment by ian-r-rose** [comment]
Looks good to me. My only comment is that I would expect `Shutdown` and `Shutdown All` to be in the same group in the menu.

**Comment by blink1073** [comment]
Sounds good, updated.

**Comment by blink1073** [comment]
@ellisonbg, should we pop up a dialog for this action, since we do so when shutting down a single kernel?

**Comment by ian-r-rose** [comment]
We actually don't show a dialog for shutting down a kernel, just for restarting or changing it. We probably should, though, since they are equally destructive.

File: packages/mainmenu-extension/src/index.ts [hunk 1]
```diff
@@ -309,6 +322,26 @@ function createKernelMenu(app: JupyterLab, menu: KernelMenu): void {
     execute: Private.delegateExecute(app, menu.kernelUsers, 'shutdownKernel')
   });
 
+  commands.addCommand(CommandIDs.shutdownAllKernels, {
+    label: 'Shutdown All Kernels',
```
Comment (ian-r-rose) [inline packages/mainmenu-extension/src/index.ts]: Now that there is a prompt for user action, can we add an ellipsis to the label?

**Pull request merged**